sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -547,6 +550,20 @@ object Radar {
             }
         })
 
+        val activityManager = this.context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            val crashLists = activityManager.getHistoricalProcessExitReasons(null, 0, 10)
+            if (crashLists.isNotEmpty()) {
+                for (crashInfo in crashLists) {
+                    if (crashInfo.reason == ApplicationExitInfo.REASON_USER_REQUESTED) {
+                        this.logger.i("User last force quit at ${dateFormat.format(Date(crashInfo.timestamp))}")
+                        break
+                    }
+                }
+            }
+        }
```

Comment:
When does `getHistoricalProcessExitReasons` get cleared?

Once we read it, we want to make sure it gets cleared / we're not duplicating these logs

I also want to better understand other `ApplicationExitInfo`'s that could exist (e.g. battery, java crashes, etc.) and which of those we should log

Reply:
https://developer.android.com/reference/android/app/ApplicationExitInfo this maybe helpful

Reply:
Indeed

so given that it's writing in a ring buffer and not clearing, I think we need to write a timestamp to user defaults when we do this check, and upon beginning this check pull that timestamp and only log the crashes before that timestamp

might as well also log the other reasons as well

Reply:
As for logging the other reason, my one concern is the lack of parity with IOS. I think we can take note of its ability to do so but revisit after the IOS implementation is done. But this is surely very useful for debugging. 

Can you elaborate a little on the need for a need to write a timestamp?

Reply:
Re: lack of parity, that makes sense. Probably better to error on the side of knowing more, but let's hold off for now.

Re: timestamp, imagine force quitting the app twice in quick succession. The first time I'd expect this implementation to send up one log. The second time, instead of just sending up one log line for the second force quit, I'd expect it'd send up two.

Reply:
Re: timestamp, the current implementation iterate through the log entries from most recent log. the break statement on line 561 should ensure that only the most recent force quit entry is sent to the dashboard.